identity-verification/module_3/3.2 Fun with Rekognition DetectFaces and SearchFacesByImage.ipynb

```diff
@@ -0,0 +1,430 @@
+   "metadata": {},
+   "source": [
+    "## Step 3. Create a new collection\n",
+    "-----\n",
+    "\n",
+    "Remember you must use a unique name if you are creating a new collection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_id='Avengers_Collection' # Remember you must use a unique name if you are creating a new collection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_collection(collection_id):\n",
+    "    #Create a collection\n",
+    "    print('Creating collection:' + collection_id)\n",
+    "    try:\n",
+    "        response=client.create_collection(CollectionId=collection_id)\n",
+    "    except:\n",
+    "        client.delete_collection(CollectionId=collection_id)\n",
+    "        response=client.create_collection(CollectionId=collection_id)\n",
+    "    print('Collection ARN: ' + response['CollectionArn'])\n",
+    "    print('Status code: ' + str(response['StatusCode']))\n",
+    "    print('Done...')\n",
+    "    \n",
+    "create_collection(collection_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 3a. Confirm your collection creation. \n",
+    "-----\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "collection_count=list_collections()\n",
+    "print(\"collections: \" + str(collection_count))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 4. Download collection images\n",
+    "---\n",
+    "here we are going to download a zip file of images, unzip it and then index it. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!rm -R avengers_face_search\n",
+    "!curl http://dz4iesrkqs7j9.cloudfront.net/random/avengers_face_search.zip -O\n",
+    "!unzip avengers_face_search.zip\n",
+    "!rm -R __MACOSX\n",
+    "!rm avengers_face_search.zip"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 5. Index faces (add faces to a collection) \n",
+    "-----\n",
+    "Here we are going to iterate over the files in the population folder and index their faces. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "directory = 'avengers_face_search/populate'\n",
+    " \n",
+    "# iterate over files in\n",
+    "# that directory\n",
+    "for filename in os.listdir(directory):\n",
+    "    f = os.path.join(directory, filename)\n",
+    "    # checking if it is a file\n",
+    "    if os.path.isfile(f):\n",
+    "        print(f)\n",
+    "        file = open(f, \"rb\") # opening for [r]eading as [b]inary\n",
```

Comment:
You should write this either using the context manager pattern (`with open(f, "rb") as file:`) or handle exceptions to ensure the file handle is closed in case of an error.

Reply:
done